src/irr/asset/CCPUMeshPacker.h

```diff
@@ -25,9 +25,21 @@ class CCPUMeshPacker final : public IMeshPacker<ICPUMeshBuffer, MDIStructType>
 
 	template <typename Iterator>
 	ReservedAllocationMeshBuffers alloc(const Iterator begin, const Iterator end);
-	virtual void commit() override;
 
-	ICPUBuffer* getMultiDrawIndirectBuffer() { return m_outMDIData.get(); }
+	//needs to be called before first commit
+	void createOutputBuffers();
+
+	template <typename Iterator>
+	PackedMeshBufferData commit(const Iterator begin, const Iterator end, ReservedAllocationMeshBuffers ramb);
```

Comment:
take the `ReservedAllocationMeshBuffers` as a reference because you'll need to put it in an invalid state.